### PR TITLE
[native] Add function to get spill path.

### PR DESCRIPTION
**Pull request by amitkdutta**
```
== NO RELEASE NOTE ==
```

### CODE REVIEW

File: presto-native-execution/presto_cpp/main/PrestoServer.h [hunk 1]
```diff
@@ -134,6 +134,9 @@ class PrestoServer {
   /// ip address.
   virtual std::string getLocalIp() const;
 
+  /// Invoked to get the spill path
```
Comment (xiaoxmeng) [inline presto-native-execution/presto_cpp/main/PrestoServer.h:137]:
```
path.
```

File: presto-native-execution/presto_cpp/main/PrestoServer.cpp [hunk 2]
```diff
@@ -313,6 +313,13 @@ void PrestoServer::run() {
 
   taskManager_->setBaseUri(taskUri);
   taskManager_->setNodeId(nodeId_);
+
+  auto baseSpillPath = getBaseSpillPath();
+  if (!baseSpillPath.empty()) {
+    taskManager_->setBaseSpillPath(baseSpillPath);
```
Comment (xiaoxmeng) [inline presto-native-execution/presto_cpp/main/PrestoServer.cpp:319]:
```
s/setBaseSpillPath/setSpillDirectory/
```

File: presto-native-execution/presto_cpp/main/TaskManager.cpp [hunk 3]
```diff
@@ -47,15 +47,15 @@ namespace {
 // the file system and sets the path to it to the Task.
 static void maybeSetupTaskSpillDirectory(
     const core::PlanFragment& planFragment,
-    exec::Task& execTask) {
-  const auto baseSpillPath = SystemConfig::instance()->spillerSpillPath();
+    exec::Task& execTask,
+    const std::string& baseSpillPath) {
```
Comment (gggrace14) [inline presto-native-execution/presto_cpp/main/TaskManager.cpp:51]: It looks to me that this change is to set and pass the baseSpillPath to maybeSetupTaskSpillDirectory() instead of getting it in maybeSetupTaskSpillDirectory() impl. If so, how does it make difference? Why this change?

File: presto-native-execution/presto_cpp/main/TaskManager.cpp [hunk 4]
```diff
@@ -47,15 +47,15 @@ namespace {
...
   const auto includeNodeInSpillPath =
       SystemConfig::instance()->includeNodeInSpillPath();
   auto nodeConfig = NodeConfig::instance();
-  if (baseSpillPath.hasValue() &&
+  if (!baseSpillPath.empty() &&
```
Comment (xiaoxmeng) [inline presto-native-execution/presto_cpp/main/TaskManager.cpp:55]:
```
if (baseSpillPath.empty() ||
      !planFragment.canSpill(execTask.queryCtx()->queryConfig()) {
   return;
}
...
```

File: presto-native-execution/presto_cpp/main/PrestoServer.cpp [hunk 5]
```diff
@@ -775,6 +778,11 @@ std::string PrestoServer::getLocalIp() const {
       "Could not infer Node IP. Please specify node.ip in the node.properties file.");
 }
 
+std::string PrestoServer::getBaseSpillPath() const {
+  const auto baseSpillPath = SystemConfig::instance()->spillerSpillPath();
+  return baseSpillPath.has_value() ? baseSpillPath.value() : "";
+}
+
```
Comment (gggrace14) [inline presto-native-execution/presto_cpp/main/PrestoServer.cpp:785]: It looks we want a default value for this config. Then https://github.com/prestodb/presto/blob/e6ae941ce7a85b42cec377f5dcc62161966605d1/presto-native-execution/presto_cpp/main/common/Configs.cpp#L244 is usually the place we specify a default value.

File: presto-native-execution/presto_cpp/main/TaskManager.h [hunk 6]
```diff
@@ -44,6 +44,10 @@ class TaskManager {
     nodeId_ = nodeId;
   }
 
+  void setBaseSpillPath(const std::string& baseSpillPath) {
+    baseSpillPath_ = baseSpillPath;
```
Comment (xiaoxmeng) [inline presto-native-execution/presto_cpp/main/TaskManager.h:48]:
```
// NOTE: we only expect to set the spill directory (or base just unify the term).
VELOX_CHECK(!spillPath.empty());
VELOX_CHECK(baseSpillPath_.empty());
```

**Review by xiaoxmeng — APPROVED** [review]
@amitkdutta thanks for looking into this and change!

**Review by gggrace14 — APPROVED** [review]
Chat about why we need this change and I'm good with it!

File: presto-native-execution/presto_cpp/main/TaskManager.h [hunk 7]
```diff
@@ -44,6 +44,11 @@ class TaskManager {
     nodeId_ = nodeId;
   }
 
+  void setBaseSpillDirectory(const std::string& baseSpillDirectory) {
+    VELOX_CHECK(!baseSpillDirectory.empty());
```
Comment (xiaoxmeng) [inline presto-native-execution/presto_cpp/main/TaskManager.h:48]: Shall we also check baseSpillDirectory_ is empty?

File: presto-native-execution/presto_cpp/main/TaskManager.h [hunk 8]
```diff
@@ -179,6 +184,7 @@ class TaskManager {
 
   std::string baseUri_;
   std::string nodeId_;
+  std::string baseSpillDirectory_;
```
Comment (xiaoxmeng) [inline presto-native-execution/presto_cpp/main/TaskManager.h:187]: nit: s/baseSpillDirectory_/baseSpillDir_/

File: presto-native-execution/presto_cpp/main/TaskManager.h (hunk 7 above)
Comment (amitkdutta) [reply presto-native-execution/presto_cpp/main/TaskManager.h:48]: We are checking if baseSpillDirectory_ is empty already. Did you mean something else?

File: presto-native-execution/presto_cpp/main/TaskManager.h (hunk 8 above)
Comment (amitkdutta) [reply presto-native-execution/presto_cpp/main/TaskManager.h:187]: Addressing in https://github.com/prestodb/presto/pull/20582
